### DISCUSSION TITLE
Regarding the matching of Smart and Smiles in GenericQueries

### DISCUSSION THREAD

**Discussion by Bellsteam** [rdkit/rdkit#7396] [Q&A]
I have written a piece of code, hoping to verify whether a Smiles or Smart containing a Genericgroup matches a Smiles structure. However, using the following code always returns false. How can I modify the code to achieve this goal?

from rdkit import Chem
smiles1 = "C1CCC(CC1)*C1=CC=CC=C1 |$;;;;;;Alkyl_p;;;;;;$|"
smiles2 = "C(CC1CCCCC1)CC1=CC=CC=C1"

mol1 = Chem.MolFromSmiles(smiles1)
mol2 = Chem.MolFromSmiles(smiles2)

if mol1 is None or mol2 is None:
    print("error")
else:
    
    qps = Chem.AdjustQueryParameters.NoAdjustments()
    qps.makeDummiesQueries = True
    sqry = Chem.AdjustQueryProperties(mol1)

    Alkyl_qry = Chem.Mol(sqry)
    Chem.SetGenericQueriesFromProperties(Alkyl_qry)

    params = Chem.SubstructMatchParameters()
    params.useGenericMatchers = True

    matches = mol2.GetSubstructMatches(Alkyl_qry, params=params)
    
    print(f"{len(matches) > 0}")

**Comment by greglandrum** [comment]
@Bellsteam Generic groups currently only work from atoms which have a single bond to them, so this case where you are trying to match an unknown number of atoms in the middle of the molecule cannot work.

One possibility for how to approximate this would be to use a link node and the RDKit's generalized substructure search features:
```
qsmarts = "[A](C1CCCCC1)c1ccccc1 |LN:0:1.5|"
smiles2 = "C(CC1CCCCC1)CC1=CC=CC=C1"

q = Chem.MolFromSmarts(qsmarts)
mol = Chem.MolFromSmiles(smiles2)

from rdkit.Chem import rdGeneralizedSubstruct
xqm = rdGeneralizedSubstruct.CreateExtendedQueryMol(q)
match = rdGeneralizedSubstruct.MolGetSubstructMatch(mol,xqm)
```
There's a bit more info about that here: https://greglandrum.github.io/rdkit-blog/posts/2023-10-17-release-new-stuff-1.html#generalized-substructure-search

If you want to indicate that the linking atoms have alkyl groups attached, you could use an SRU group:
```
polyq = Chem.MolFromSmarts('*C(C1CCCCC1)c1ccccc1 |$Alkyl_p;;;;;;;;;;;;;$,Sg:n:1,0:1-5:ht|')
smiles3 = "CC(C(C)C1CCCCC1)C(C)C1=CC=CC=C1"
mol3 = Chem.MolFromSmiles(smiles3)
xqm2 = rdGeneralizedSubstruct.CreateExtendedQueryMol(polyq)
match = rdGeneralizedSubstruct.MolGetSubstructMatch(mol3,xqm2)
```
But that will not work if you don't have extra substituents on the alkyl chains

**Reply by Bellsteam** [reply]
THX. I will try this method
